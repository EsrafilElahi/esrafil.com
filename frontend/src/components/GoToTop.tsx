import { useEffect, useState } from 'react';
import Arrow from '/public/icons/arrow.svg?react';

const GoToTop = () => {
  const [scrollY, setScrollY] = useState<number>(0);

  const goToTop = () => {
    window.scrollTo({ top: 0, behavior: 'smooth' });
  };

  useEffect(() => {
    const handleScroll = () => {
      const scrollY = window.scrollY || window.pageYOffset;
      setScrollY(scrollY);
    };

    // Add event listener for scroll events
    window.addEventListener('scroll', handleScroll);

    // Cleanup the event listener on component unmount
    return () => {
      window.removeEventListener('scroll', handleScroll);
    };
  }, []);

  return (
    <Arrow
      role='img'
      className={`${scrollY < 415 && 'hidden'} w-9 h-9 cursor-pointer fixed bottom-[8%] md:bottom-[11%] right-10`}
      onClick={goToTop}
    />
  );
};

export default GoToTop;
