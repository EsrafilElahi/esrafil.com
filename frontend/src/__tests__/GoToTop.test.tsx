import { cleanup, fireEvent, render, screen, waitFor, within } from '@testing-library/react';
import { afterEach, beforeEach, describe, test } from 'vitest';
import GoToTop from '../components/GoToTop';

beforeEach(() => {
  render(<GoToTop />);
});

afterEach(() => {
  cleanup();
});

describe('GoToTop Component', () => {
  test('render GoToTop', () => {
    const { container } = render(<GoToTop />);
    expect(container).toBeInTheDocument();
  });

  test('GoToTop snapshot', () => {
    const result = render(<GoToTop />);
    expect(result).toMatchSnapshot();
  });

  test('exist and click on it', async () => {
    const arrow = screen.getByRole('img');
    expect(arrow).toBeInTheDocument();
  });
});
