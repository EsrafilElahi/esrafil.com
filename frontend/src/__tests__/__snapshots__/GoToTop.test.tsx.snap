// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GoToTop Component > GoToTop snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        class="hidden w-9 h-9 cursor-pointer fixed bottom-[8%] md:bottom-[11%] right-10"
        role="img"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Artboard-28
        </title>
        <g
          data-name="Circle Up"
          id="Circle_Up-2"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
            style="fill: #ece4b7;"
          />
          <path
            d="M12.707,6.293a1,1,0,0,0-1.414,0l-3,3a1,1,0,0,0,1.414,1.414L11,9.414V17a1,1,0,0,0,2,0V9.414l1.293,1.293a1,1,0,0,0,1.414-1.414Z"
            style="fill: #ff8e31;"
          />
        </g>
      </svg>
    </div>
    <div>
      <svg
        class="hidden w-9 h-9 cursor-pointer fixed bottom-[8%] md:bottom-[11%] right-10"
        role="img"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Artboard-28
        </title>
        <g
          data-name="Circle Up"
          id="Circle_Up-2"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
            style="fill: #ece4b7;"
          />
          <path
            d="M12.707,6.293a1,1,0,0,0-1.414,0l-3,3a1,1,0,0,0,1.414,1.414L11,9.414V17a1,1,0,0,0,2,0V9.414l1.293,1.293a1,1,0,0,0,1.414-1.414Z"
            style="fill: #ff8e31;"
          />
        </g>
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      class="hidden w-9 h-9 cursor-pointer fixed bottom-[8%] md:bottom-[11%] right-10"
      role="img"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Artboard-28
      </title>
      <g
        data-name="Circle Up"
        id="Circle_Up-2"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
          style="fill: #ece4b7;"
        />
        <path
          d="M12.707,6.293a1,1,0,0,0-1.414,0l-3,3a1,1,0,0,0,1.414,1.414L11,9.414V17a1,1,0,0,0,2,0V9.414l1.293,1.293a1,1,0,0,0,1.414-1.414Z"
          style="fill: #ff8e31;"
        />
      </g>
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
